src/core/common/timer.hpp

```diff
@@ -181,6 +181,47 @@ class TimerMilli : public Timer
      */
     void Stop(void);
 
+    /**
+     * This static method returns the time diff in milliseconds between @p aTime1 and @p aTime2.
+     *
+     * @param[in]   aStart  The start time.
+     * @param[in]   aEnd    The end time.
+     *
+     * @returns The time diff in milliseconds.
+     *
+     */
+    static int32_t Diff(uint32_t aStart, uint32_t aEnd) { return static_cast<int32_t>(aEnd - aStart); }
```

Comment:
Highly suggest for us to be careful about casting `uint32` to `int32` and avoid using it when we can (specially when the `uint32_t` value can be larger than `2^31`). While I expect this to work (as we expect it) on all compilers/platforms, from C++ spec, converting from `uint32_t` to a signed `int32_t` when the value does not fit in the signed type is not clearly defined and is left to compiler/implementation.

> If the destination type is signed, the value does not change if the source integer can be represented in the destination type. Otherwise the result is implementation-defined (until C++20)
[https://en.cppreference.com/w/cpp/language/implicit_conversion]

We handle this in `bool TimerScheduler::IsStrictlyBefore(uint32_t aTimeA, uint32_t aTimeB)`.

Reply:
Thanks for the information, I updated to use the [type alias](https://en.cppreference.com/w/cpp/language/reinterpret_cast). It seems to be safe.